mlir/include/mlir/Dialect/Mesh/IR/MeshOps.td

```diff
@@ -196,16 +197,18 @@ def Mesh_ShardingOp : Mesh_Op<"sharding", [
     e.g. the first sharded dimension gets [1,2] halos and the seconds gets [2,3] halos.
     `?` indicates dynamic halo sizes.
     
-    6. [Optional] Sizes of sharded dimensions of each shard.
-    `sharded_dims_sizes`is provided as a flattened 1d array of i64s: for each device of the
-    device-mesh one value for each sharded tensor dimension.
+    6. [Optional] Offsets for each shard and sharded tensor dimension.
+    `sharded_dims_offsets` is provided as a flattened 1d array of i64s:
```

Comment:
Yes, ideally this would be a 2d array. As we allow dynamic and static values, I didn't intuitively know how to implement the parsing for it. Mechanics exist for 1d arrays, though. We can add this in an extra PR, but at this point I didn't want to spend time on it. Of course it'd be great if someone had a solution.

Reply:
I think for static values, we can use IndexListArrayAttr or SmallVector<SmallVector<int64_t>>. For dynamic values, we can use SmallVector<SmallVector<Value>>. But I'm totally fine that we can optimize it later.

With regarding to the check, we can add the check logic in the `verify` function only for static values.

Reply:
added a check and tests